### PR TITLE
Resource "File" correctly uses scmID

### LINKED ISSUE #558: Resource File is failing when used with scmID
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The following example:

```
scms:
  scenario-1:
    kind: git
    spec:
      url: "git@github.com:updatecli/updatecli.git"
      branch: "main"

sources:
  scenario-1:
    name: "Scenario 1"
    kind: file
    scmID: scenario-1
    spec:
      file: README.md
      line: 1

```

fails with error message

```
################
# TEST GIT SCM #
################


SOURCES
=======
Validating "GIT.YAML"

default
-------
ERROR: ✗ open README.md: no such file or directory

```

### Expected Behavior

It should correctly find the FILE from the temporary location like for condition and targets

### Steps To Reproduce

Just run updatecli 

### Environment

```markdown
- OS:
- updatecli:
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by olblak**
Fix #558 

If a relative file path is provided with a scmID, then like for condition and target, we want to use the git repository directory as the file root directory

## Test

To test this pull request, I used the following updatecli configuration:

.updatecli.yaml
```
---
title: "Test Git scm"

scms:
  scenario-1:
    kind: git 
    spec:
      username: '{{ requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'
      password: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
      url: "https://github.com/updatecli/experiment.git"
      branch: "main"

sources:
  scenario-1:
    name: "Scenario 1"
    kind: file
    scmID: scenario-1
    spec:
      file: README.md
      line: 1
```

```shell
updatecli diff --config updatecli.yaml
```

### Tradeoff
/
### Potential improvement

/

### CODE REVIEW

File: pkg/plugins/resources/file/source.go [hunk 1]
```diff
@@ -27,6 +28,11 @@ func (f *File) Source(workingDir string) (string, error) {
 		return "", fmt.Errorf("Validation error: the provided manifest configuration had the following validation errors:\n%s", strings.Join(validationErrors, "\n\n"))
 	}
 
+	if !filepath.IsAbs(f.spec.File) {
+		f.spec.File = filepath.Join(workingDir, f.spec.File)
+		logrus.Debugf("Relative path detected: changing to absolute path from working directory: %q", f.spec.File)
+	}
```
Comment (dduportal) [inline pkg/plugins/resources/file/source.go:34]: This code block have to be associated withg a unit test :)

File: pkg/plugins/resources/file/source.go [hunk 2]
```diff
@@ -27,6 +28,11 @@ func (f *File) Source(workingDir string) (string, error) {
 		return "", fmt.Errorf("Validation error: the provided manifest configuration had the following validation errors:\n%s", strings.Join(validationErrors, "\n\n"))
 	}
 
+	if !filepath.IsAbs(f.spec.File) {
```
Comment (dduportal) [inline pkg/plugins/resources/file/source.go:31]:
```suggestion
        # Relative path is used when an SCM is associated with the file resource: means the file is on a remote SCM (hence relative path)
	if !filepath.IsAbs(f.spec.File) {
```

**Review by dduportal — CHANGES_REQUESTED** [review]
Good catch!

Can you:
- rebase against master (to only embed the commits related to this PR). easier to review.
- Add unit test(s) to cover the new code block
?

File: pkg/plugins/resources/file/source.go [hunk 3]
```diff
@@ -27,6 +28,12 @@ func (f *File) Source(workingDir string) (string, error) {
 		return "", fmt.Errorf("Validation error: the provided manifest configuration had the following validation errors:\n%s", strings.Join(validationErrors, "\n\n"))
 	}
 
+        # Relative path is used when an SCM is associated with the file resource: means the file is on a remote SCM (hence relative path)
```
Comment (dduportal) [inline pkg/plugins/resources/file/source.go:31]:
```suggestion
        // Relative path is used when an SCM is associated with the file resource: means the file is on a remote SCM (hence relative path)
```

My bad, E_TOO_MUCH_GROOVY

**Comment by olblak** [comment]
@dduportal I had to slightly modify the test struct

**Review by dduportal — APPROVED** [review]
@olblak you did not "slightly change" the test struct: you fixed it as it was incorrect (I don't know how did I mess it up, but it's fix with your change). Thanks!

**Comment by olblak** [comment]
> @olblak you did not "slightly change" the test struct: you fixed it as it was incorrect (I don't know how did I mess it up, but it's fix with your change). Thanks!

Well, you did a great job for those UT, that was just a minor mistake which to be honest confused me initially :D

**Pull request merged**